src/sheet-common.tsx

```diff
@@ -0,0 +1,393 @@
+'use client';
+/*
+ * Common code shared by web sheet components.
+ *
+ * IMPORTANT: Do NOT import anything from here into `sheet-root.tsx`
```

Comment:
Maybe you can explain why in this comment. Otherwise, in a few months someone may read this code and they won't understand the reason behind this

Reply:
ok